tests/test_cuda_bd_exchange_mover.py

```diff
@@ -16,10 +18,77 @@
 from timemachine.md import builders
 from timemachine.md.barostat.utils import get_bond_list, get_group_indices
 from timemachine.md.exchange.exchange_mover import BDExchangeMove as RefBDExchangeMove
-from timemachine.potentials import HarmonicBond, Nonbonded
+from timemachine.md.minimizer import check_force_norm
+from timemachine.potentials import HarmonicBond, Nonbonded, SummedPotential
 from timemachine.testsystems.relative import get_hif2a_ligand_pair_single_topology
 
 
+def verify_bias_deletion_moves(
+    mol_groups: List,
+    bdem,
+    ref_bdem: RefBDExchangeMove,
+    conf: NDArray,
+    box: NDArray,
+    total_num_proposals: int,
+    proposals_per_move: int,
+    rtol: float,
+    atol: float,
+):
+    assert bdem.last_log_probability() == 0.0, "First log probability expected to be zero"
+    accepted = 0
+    last_conf = conf
+    for step in range(total_num_proposals // proposals_per_move):
+        x_move, x_box = bdem.move(last_conf, box)
+        # The box will never change
+        np.testing.assert_array_equal(box, x_box)
+        num_moved = 0
+        new_pos = None
+        idx = -1
+        for i, mol_idxs in enumerate(ref_bdem.water_idxs_np):
+            if not np.all(x_move[mol_idxs] == last_conf[mol_idxs]):
+                num_moved += 1
+                new_pos = x_move[mol_idxs]
+                idx = i
+        if num_moved > 0:
+            accepted += 1
+        if num_moved == 1:
+            # The molecules should all be imaged in the home box
+            np.testing.assert_allclose(image_frame(mol_groups, x_move, x_box), x_move)
```

Comment:
Improves the ability to test by at least verifying that when a single move was accepted within a series of proposals that the reference platform says that the move has a non-zero probability. This was how I caught the lack of W support causing issues.

Reply:
If `num_moves > 1` then we don't run these tests?

Reply:
Hm that seems non-ideal, since a test can just by pure chance skip this block (if it happened to accept multiple moves).

Reply:
Hm thinking about this some more - does the ordering matter here? I.e. suppose have 5 waters, and we waters 2 and 4 are moved. If we simulate moving 2 then 4 vs moving 4 then 2, shouldn't the final log weights be path independent?

Reply:
The final log weights will be path independent, so as you said in https://github.com/proteneer/timemachine/pull/1252#discussion_r1497905782 you could compute the after log weights and the probability should still be > 0.0.

Not sure this is true of the targeted case where the probability is impacted by the number of waters in each region